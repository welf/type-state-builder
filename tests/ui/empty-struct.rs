use type_state_builder::TypeStateBuilder;

// This should be an error: empty structs are not supported
#[derive(TypeStateBuilder)]
struct EmptyStruct {}

fn main() {}
