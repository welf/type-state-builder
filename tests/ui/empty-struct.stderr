error: Struct 'EmptyStruct' has no fields
       note: builder generation requires at least one field to be meaningful
       help: add some fields to your struct
 --> tests/ui/empty-struct.rs:4:10
  |
4 | #[derive(TypeStateBuilder)]
  |          ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `TypeStateBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
